### PR TITLE
Add bash script to install common depednecies, update version of Go everywhere to 1.20.1

### PR DESCRIPTION
**Pull request by andrewhop**
### Issues:
Addresses CryptoAlg-1439, CryptoAlg-1532

### Description of changes: 
Update the version of go in our CI, this unblocks changes that need a newer version of Go. I decided to move the go installation instructions to a common script per platform in the hopes to reduce the copy paste in the dockerfiles. This does mean each dockerfile has to have new common code to copy and install the programs but I'm hoping future changes can use this pattern, such as cmake.

### Testing:
Built and ran images locally

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and 
the ISC license.

### CODE REVIEW

File: tests/ci/docker_images/linux-aarch/build_images.sh [hunk 1]
```diff
@@ -6,22 +6,22 @@
 TOKEN=$(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:ratelimitpreview/test:pull" | jq -r .token)
 curl --head -H "Authorization: Bearer $TOKEN" https://registry-1.docker.io/v2/ratelimitpreview/test/manifests/latest
 
-docker build -t amazonlinux-2-aarch:base amazonlinux-2_base
+docker build -t amazonlinux-2-aarch:base -f amazonlinux-2_base/Dockerfile .
```
Comment (dkostic) [inline tests/ci/docker_images/linux-aarch/build_images.sh:9]: why do we need the change only for some docker files?

File: tests/ci/docker_images/linux-x86/centos-8_gcc-8x/Dockerfile [hunk 2]
```diff
@@ -21,13 +23,15 @@ RUN set -ex && \
...
     ninja-build \
     perl \
-    golang \
     wget \
     unzip && \
     # Based on https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2-linux.html
     curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" && \
     unzip awscliv2.zip && \
     ./aws/install --bin-dir /usr/bin && \
+    wget https://dl.google.com/go/go1.20.1.linux-amd64.tar.gz && \
```
Comment (dkostic) [inline tests/ci/docker_images/linux-x86/centos-8_gcc-8x/Dockerfile:32]: do we need these 3 lines even with the new script called below?

File: tests/ci/docker_images/linux-x86/ubuntu-20.04_android/Dockerfile [hunk 3]
```diff
@@ -29,8 +29,8 @@ RUN set -ex && \
     unzip && \
 # install go dependencies
     cd /tmp && \
-    wget https://dl.google.com/go/go1.13.12.linux-amd64.tar.gz && \
-    tar -xvf go1.13.12.linux-amd64.tar.gz && \
+    wget https://dl.google.com/go/go1.20.1.linux-amd64.tar.gz && \
```
Comment (dkostic) [inline tests/ci/docker_images/linux-x86/ubuntu-20.04_android/Dockerfile:32]: we can't use the new script for this Dockerfile?

File: tests/ci/docker_images/linux-x86/ubuntu-22.04_base/Dockerfile [hunk 4]
```diff
@@ -43,8 +43,8 @@ RUN set -ex && \
     git clone https://github.com/llvm/llvm-project.git --branch llvmorg-11.1.0  --depth 1 && \
     cd llvm-project && rm -rf $(ls -A | grep -Ev "(libcxx|libcxxabi)") && \
     cd /tmp && \
-    wget https://dl.google.com/go/go1.13.12.linux-amd64.tar.gz && \
-    tar -xvf go1.13.12.linux-amd64.tar.gz && \
+    wget https://dl.google.com/go/go1.20.1.linux-amd64.tar.gz && \
```
Comment (dkostic) [inline tests/ci/docker_images/linux-x86/ubuntu-22.04_base/Dockerfile:46]: we can't use the new script for this Dockerfile?

File: tests/ci/docker_images/rust/linux-386/Dockerfile [hunk 5]
```diff
@@ -6,7 +6,10 @@ FROM --platform=linux/386 rust:latest
 SHELL ["/bin/bash", "-c"]
 
 RUN apt-get update && \
-    apt-get -y install clang cmake golang && \
+    apt-get -y install clang cmake && \
+    wget https://dl.google.com/go/go1.20.1.linux-amd64.tar.gz && \
```
Comment (dkostic) [inline tests/ci/docker_images/rust/linux-386/Dockerfile:10]: we can't use the new script for this Dockerfile?

File: tests/ci/docker_images/rust/linux-arm64/Dockerfile [hunk 6]
```diff
@@ -10,7 +10,10 @@ FROM --platform=linux/arm64 rust:latest
 SHELL ["/bin/bash", "-c"]
 
 RUN apt-get update && \
-    apt-get -y install clang cmake golang && \
+    apt-get -y install clang cmake && \
+    wget https://dl.google.com/go/go1.20.1.linux-amd64.tar.gz && \
```
Comment (dkostic) [inline tests/ci/docker_images/rust/linux-arm64/Dockerfile:14]: we can't use the new script for this Dockerfile?

File: tests/ci/docker_images/rust/linux-x86_64/Dockerfile [hunk 7]
```diff
@@ -6,7 +6,10 @@ FROM --platform=linux/amd64 rust:latest
 SHELL ["/bin/bash", "-c"]
 
 RUN apt-get update && \
-    apt-get -y install clang cmake golang && \
+    apt-get -y install clang cmake && \
+    wget https://dl.google.com/go/go1.20.1.linux-amd64.tar.gz && \
```
Comment (dkostic) [inline tests/ci/docker_images/rust/linux-x86_64/Dockerfile:10]: we can't use the new script for this Dockerfile?

**Review by samuel40791765 — APPROVED** [review]

**Review by dkostic — APPROVED** [review]

**Review by samuel40791765 — APPROVED** [review]

**Pull request merged**